lib/init.coffee

```diff
@@ -103,7 +103,41 @@ module.exports =
         reResult = re.exec(toParse)
       ret
 
+    # only elixir 1.3+
     parseWarning = (toParse, textEditor) ->
+      console.log('to parse', toParse)
+
+      ret = []
+      re = ///
+        warning:\ (.*)\n # warning
+        \ \ (.*):([0-9]+) # file and file number
+        ///g
+      reResult = re.exec(toParse)
+
+      while reResult?
+        console.log(reResult)
+        try
+          ret.push
+            type: "Warning"
+            text: reResult[1]
+            filePath: path.join(projectPath(textEditor), reResult[2])
+            # use range, this because the previous method of getting a range
+            # used the current buffer that is open in the texteditor, and it
+            # blows up if the line number is larger than the current file
+            # if the compiler returned warnings or errors of other files,
+            # it would thus frequently blow up and return no errors/warnings
+            # at all
+            range: new Range(
```

Comment:
We don't _need_ to bring in Range here btw, a "Range compatible array" works just fine. It doesn't hurt anything to do it this way though :wink:.